streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1820,6 +1820,7 @@ void addRecordsToTasks(final ConsumerRecords<byte[], byte[]> records) {
             }
 
             activeTask.addRecords(partition, records.records(partition));
+            activeTask.setNextOffsetsAndMetadata(records.nextOffsets());
```

Comment:
I am wondering if, instead of setting, an update should be done. Then, instead of overwriting the former map, we just update it by adding new entries or updating the existing ones.

Reply:
Would it be better to not add `setNextOffsetsAndMetadata` but make it part of `addRecord` to begin with?

In the end, we only need to pass in `nextRecord.offset` for the `partition` we are currently processing, plus a leader epoch, but not the full map of `nextRecords()`?

Reply:
I agree:)